spring-restdocs-asciidoctor/src/test/java/org/springframework/restdocs/asciidoctor/RestDocsIncludeBlockMacroTest.java

```diff
@@ -0,0 +1,165 @@
+ * Because there is no java implementation (yet)
+ * we can only test the behaviour when rendering
+ *
+ * @author Gerrit Meier
+ */
+public class RestDocsIncludeBlockMacroTest {
+
+	private final Options options = new Options();
+	private final Asciidoctor asciidoctor = Asciidoctor.Factory.create();
+
+	@Before
+	public void setUp() {
+		this.options.setAttributes(getAttributes());
+	}
+
+	private Attributes getAttributes() {
+		Attributes attributes = new Attributes();
+		attributes.setAttribute("projectdir", new File(".").getAbsolutePath());
+		return attributes;
+	}
+
+	@Test
+	public void includeSnippetWithLevel() {
+		String expectedContent = "<div class=\"sect1\">\n" +
+				"<h2 id=\"_curl_request\">curl request</h2>\n" +
+				"<div class=\"sectionbody\">\n" +
+				"<div class=\"listingblock\">\n" +
+				"<div class=\"content\">\n" +
+				"<pre class=\"highlight\"><code class=\"language-bash\" data-lang=\"bash\">$ curl 'http://localhost:8080/' -i</code></pre>\n" +
+				"</div>\n" +
+				"</div>\n" +
+				"</div>\n" +
+				"</div>";
+
+		String result = this.asciidoctor.convert(
+				"srd::some-operation[snippets='curl-request', level=2]", this.options);
+
+		assertThat(result, equalTo(expectedContent));
+	}
+
+	@Test
+	public void includeMultipleSnippets() {
+		String expectedContent = "<div class=\"sect1\">\n" +
+				"<h2 id=\"_curl_request\">curl request</h2>\n" +
+				"<div class=\"sectionbody\">\n" +
+				"<div class=\"listingblock\">\n" +
+				"<div class=\"content\">\n" +
+				"<pre class=\"highlight\"><code class=\"language-bash\" data-lang=\"bash\">$ curl 'http://localhost:8080/' -i</code></pre>\n" +
+				"</div>\n" +
+				"</div>\n" +
+				"</div>\n" +
+				"</div>\n" +
+				"<div class=\"sect1\">\n" +
+				"<h2 id=\"_http_request\">HTTP request</h2>\n" +
+				"<div class=\"sectionbody\">\n" +
+				"<div class=\"listingblock\">\n" +
+				"<div class=\"content\">\n" +
+				"<pre class=\"highlight nowrap\"><code class=\"language-http\" data-lang=\"http\">GET / HTTP/1.1\n" +
+				"Host: localhost:8080</code></pre>\n" +
+				"</div>\n" +
+				"</div>\n" +
+				"</div>\n" +
+				"</div>";
+
+		String result = this.asciidoctor.convert(
+				"srd::some-operation[snippets='curl-request,http-request', level=2]", this.options);
+
+		assertThat(result, equalTo(expectedContent));
+	}
+
+	@Test
+	public void includeSnippetWithDefaultLevel() {
+		String expectedContent = "<div class=\"sect3\">\n" +
+				"<h4 id=\"_curl_request\">curl request</h4>\n" +
+				"<div class=\"listingblock\">\n" +
+				"<div class=\"content\">\n" +
+				"<pre class=\"highlight\"><code class=\"language-bash\" data-lang=\"bash\">$ curl 'http://localhost:8080/' -i</code></pre>\n" +
+				"</div>\n" +
+				"</div>\n" +
+				"</div>";
+
+		String result = this.asciidoctor.convert(
+				"srd::some-operation[snippets='curl-request']", this.options);
+
+		assertThat(result, equalTo(expectedContent));
+	}
+
+	@Test
+	public void useMacroWithoutSnippetAddsNothing() {
```

Comment:
I'm intrigued by this test. What was the reasoning behind adding nothing? I would expect this to either produce an error or to include all of the snippets for a particular operation. My preference is for the latter but I'm not set on that.